components/director/internal/domain/application/repository_test.go

```diff
@@ -15,39 +20,319 @@ import (
+		pgRepository := application.NewPostgresRepository(conv)
+		//THEN
+		_, err := pgRepository.List(ctx, tenantID, nil, 2, pagination.EncodeNextOffsetCursor(1, -3))
+
+		//THEN
+		require.EqualError(t, err, "while decoding page cursor: cursor is not correct")
+	})
+}
+
+func TestPgRepository_GetByID_ShouldReturnApplicationModelForApplicationEntity(t *testing.T) {
+	// given
+	conv := application.NewConverter(nil, nil, nil, nil)
+	tenantID := uuid.New().String()
+	appID := uuid.New().String()
+
+	timestamp, err := time.Parse(time.RFC3339, "2002-10-02T10:00:00-05:00")
+	require.NoError(t, err)
+
+	sqlxDB, sqlMock := testdb.MockDatabase(t)
+	defer sqlMock.AssertExpectations(t)
+
+	rows := sqlmock.NewRows([]string{"id", "tenant_id", "name", "description", "status_condition", "status_timestamp", "healthcheck_url"}).
+		AddRow(appID, tenantID, "App ABC", "Description for application ABC", "INITIAL", timestamp, "http://domain.local")
+
+	sqlMock.ExpectQuery(`^SELECT (.+) FROM public.applications WHERE tenant_id = \$1 AND id = \$2$`).
+		WithArgs(tenantID, appID).
+		WillReturnRows(rows)
+
+	ctx := persistence.SaveToContext(context.TODO(), sqlxDB)
+
+	pgRepository := application.NewPostgresRepository(conv)
+
+	// when
+	modelApp, err := pgRepository.GetByID(ctx, tenantID, appID)
+
+	//then
+	require.NoError(t, err)
+	assert.NoError(t, sqlMock.ExpectationsWereMet())
+	assert.Equal(t, appID, modelApp.ID)
+	assert.Equal(t, tenantID, modelApp.Tenant)
+}
+
+func TestPgRepository_Create_ShouldCreateApplicationEntityFromValidModel(t *testing.T) {
+	// given
+	appID := uuid.New().String()
+	tenantID := uuid.New().String()
+	timestamp, err := time.Parse(time.RFC3339, "2002-10-02T10:00:00-05:00")
+	assert.NoError(t, err)
+
+	description := "Description for application BCD"
+	healthcheckURL := "http://domain.local"
+	modelApp := &model.Application{
+		ID:          appID,
+		Tenant:      tenantID,
+		Name:        "Application XYZ",
+		Description: &description,
+		Status: &model.ApplicationStatus{
+			Condition: model.ApplicationStatusConditionInitial,
+			Timestamp: timestamp,
+		},
+		HealthCheckURL: &healthcheckURL,
+	}
+
+	sqlxDB, sqlMock := testdb.MockDatabase(t)
+	defer sqlMock.AssertExpectations(t)
+
+	sqlMock.ExpectExec(regexp.QuoteMeta(`INSERT INTO public.applications ( id, tenant_id, name, description, status_condition, status_timestamp, healthcheck_url ) VALUES ( ?, ?, ?, ?, ?, ?, ? )`)).
+		WithArgs(modelApp.ID, modelApp.Tenant, modelApp.Name, modelApp.Description, modelApp.Status.Condition, modelApp.Status.Timestamp, healthcheckURL).
+		WillReturnResult(sqlmock.NewResult(-1, 1))
+
+	ctx := persistence.SaveToContext(context.TODO(), sqlxDB)
+
+	conv := application.NewConverter(nil, nil, nil, nil)
+
+	pgRepository := application.NewPostgresRepository(conv)
+
+	// when
+	err = pgRepository.Create(ctx, modelApp)
+
+	// then
+	assert.NoError(t, err)
+}
+
+func TestPgRepository_Update_ShouldUpdateApplicationEntityFromValidModel(t *testing.T) {
```

Comment:
Test the following cases:
- Success
- Converter Error
- DB Error

See webhook repository for reference.